docs/notes/canvas-id-parsing.md

```diff
@@ -0,0 +1,71 @@
+# Canvas Id parsing
+
+When parsing payloads, the `canvasId` can be used to generate a `canvasPainting` `id` record.  This can be accepted either directly from a `canvasPainting` `canvasId` and also from `items` `id` of a canvas.  In addition to the 2 locations that we can generate a `canvasId` from, we can accept either a short form (i.e.: just the id itself) or alternatively from a recognized URI.  Finally, this `canvasId` can be used to join a `canvas` declared in `items` with a `canvasPainting` record in order to decorate an asset from the DLCS with additional IIIF.   This document is set to explain the various ways this `canvasId` can be parsed from a payload.
+
+## Accepted formats
+
+As mentioned above, there are a few formats that will be accepted 
+
+
+| Format      | Example     |
+| ------------- | ------------- |
+| short canvas | `someId` |
+| general API URL | `https://presentation-api.com/1/canvas/someId` |
+| customer specific API URL | `https:/customer-base.com/canvas/someId` |
+
+
+## Basic process
+
+When we receive a canvas id, a series of actions happen to see if we can work out the canvas id based on the given value.  This can be quite complex to follow the logic, so the below flowchart shows the steps taken:
+
+```mermaid
+flowchart TD
+    first[Canvas provided]
+    second{is the canvas id null}
+    third[generate the canvas id]
+    fourth{is the canvas id a URI}
+    fifth{any invalid characters}
+    sixth[use the provided canvas id]
+    seventh{where is this canvas id from?}
+    eighth{is this a recognized host}
+    ninth[parse path with rewrites]
+    tenth{is the path parsed}
+    eleventh{from items}
+    twelfth[throw an error]
+
+    first --> second
+    second -- yes --> third
+    second -- no --> fourth
+    fourth -- no --> fifth
+    fifth -- yes --> eleventh
+    fifth -- no --> sixth
+    fourth -- yes --> seventh
+    seventh -- items --> eighth
+    eighth -- no --> third
+    eighth -- yes --> ninth
+    seventh -- canvasPaintings --> ninth
+    ninth --> tenth
+    tenth -- no --> third
```

Comment:
If we receive `canvasPainting.canvasId` and can't parse it shouldn't it be rejected, rather than us creating an alternative id?

Reply:
This is going down the PaintedResources path though (unless I'm reading this wrong)

```
where is this canvas id from? -> canvasPaintings -> parse path with rewrites -> is the path parsed -> no`

seventh -- canvasPaintings --> ninth --> tenth --> no                       
```